### PR TITLE
feat(YT Music - Hide 'Get Music Premium' label): Remove occurences of label in settings

### LINKED ISSUE #2984: bug (YT Music): Patches fails on version 6.46.52+
### Bug description

When patching the latest version of YT Music.(6.46.52) the "Hide 'Get Music Premium' label' fails to apply.

Same with the "Remove upgrade button" patch

### Error logs

```shell
Hide 'Get Music Premium' label failed: app.revanced.patcher.patch.PatchException: Failed to resolve HideGetPremiumFingerprint
	at app.revanced.util.BytecodeUtilsKt.getException(BytecodeUtils.kt:27)
	at app.revanced.patches.music.layout.premium.HideGetPremiumPatch.execute(HideGetPremiumPatch.kt:43)
	at app.revanced.patches.music.layout.premium.HideGetPremiumPatch.execute(HideGetPremiumPatch.kt:14)
	at app.revanced.patcher.Patcher$apply$1.invokeSuspend$executePatch(Unknown Source:190)
	at app.revanced.patcher.Patcher$apply$1.invokeSuspend(Unknown Source:292)
	at app.revanced.patcher.Patcher$apply$1.invoke(SourceFile:1)
	at app.revanced.patcher.Patcher$apply$1.invoke(SourceFile:2)
	at b6.f.b(Unknown Source:2)
	at b6.a.a(Unknown Source:71)
	at app.revanced.manager.flutter.MainActivity$runPatcher$1$patcherResult$1$2.invokeSuspend(Unknown Source:76)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(Unknown Source:11)
	at y5.o0.run(Unknown Source:93)
	at y5.u0.P(Unknown Source:46)
	at y5.e.t0(Unknown Source:23)
	at y5.h.a(Unknown Source:73)
	at y5.g.c(Unknown Source:0)
	at y5.h.b(Unknown Source:6)
	at y5.g.d(Unknown Source:0)
	at app.revanced.manager.flutter.MainActivity.runPatcher$lambda$34(Unknown Source:470)
	at app.revanced.manager.flutter.MainActivity.k(Unknown Source:0)
	at app.revanced.manager.flutter.f.run(Unknown Source:20)
	at java.lang.Thread.run(Thread.java:1012)
Remove upgrade button failed: app.revanced.patcher.patch.PatchException: Failed to resolve PivotBarConstructorFingerprint
	at app.revanced.util.BytecodeUtilsKt.getException(BytecodeUtils.kt:27)
	at app.revanced.patches.music.layout.upgradebutton.RemoveUpgradeButtonPatch.execute(RemoveUpgradeButtonPatch.kt:82)
	at app.revanced.patches.music.layout.upgradebutton.RemoveUpgradeButtonPatch.execute(RemoveUpgradeButtonPatch.kt:22)
	at app.revanced.patcher.Patcher$apply$1.invokeSuspend$executePatch(Unknown Source:190)
	at app.revanced.patcher.Patcher$apply$1.invokeSuspend(Unknown Source:292)
	at app.revanced.patcher.Patcher$apply$1.invoke(SourceFile:1)
	at app.revanced.patcher.Patcher$apply$1.invoke(SourceFile:2)
	at b6.f.b(Unknown Source:2)
	at b6.a.a(Unknown Source:71)
	at app.revanced.manager.flutter.MainActivity$runPatcher$1$patcherResult$1$2.invokeSuspend(Unknown Source:76)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(Unknown Source:11)
	at y5.o0.run(Unknown Source:93)
	at y5.u0.P(Unknown Source:46)
	at y5.e.t0(Unknown Source:23)
	at y5.h.a(Unknown Source:73)
	at y5.g.c(Unknown Source:0)
	at y5.h.b(Unknown Source:6)
	at y5.g.d(Unknown Source:0)
	at app.revanced.manager.flutter.MainActivity.runPatcher$lambda$34(Unknown Source:470)
	at app.revanced.manager.flutter.MainActivity.k(Unknown Source:0)
	at app.revanced.manager.flutter.f.run(Unknown Source:20)
	at java.lang.Thread.run(Thread.java:1012)
Compiling patched dex files
Compiled 8 dex files
Compiling modified resources
Aligning APK
Signing APK
Patched APK
```

### Solution

_No response_

### Additional context

_No response_

### Acknowledgements

- [X] This issue is not a duplicate of an existing bug report.
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.

### PR DESCRIPTION
**Pull request by ponces**
Fixes #2984.
Credits to @inotia00.

### CODE REVIEW

File: src/main/kotlin/app/revanced/patches/music/layout/premium/HideGetPremiumPatch.kt [hunk 1]
```diff
@@ -41,5 +57,57 @@ object HideGetPremiumPatch : BytecodePatch(
...
+                val targetIndex = getWideLiteralInstructionIndex(HideGetPremiumResourcePatch.privacyTosFooterId) + 4
+                targetReference = getInstruction<ReferenceInstruction>(targetIndex + 1).reference
+
+                with(
+                    context
+                        .toMethodWalker(this)
+                        .nextMethod(targetIndex, true)
+                        .getMethod() as MutableMethod
+                ) {
+                    this.implementation!!.instructions.apply {
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/music/layout/premium/HideGetPremiumPatch.kt:72]: This implementation needs to be refactored, ideally to a fingerprint with an opcode pattern like it used to be before. 
Such implementation as it is right now should be a last resort.

File: src/main/kotlin/app/revanced/patches/music/layout/premium/HideGetPremiumPatch.kt [hunk 2]
```diff
@@ -41,5 +57,57 @@ object HideGetPremiumPatch : BytecodePatch(
...
+                                break
+                            }
+                        }
+                    }
+                }
+            }
+        } ?: throw AccountMenuFooterFingerprint.exception
+
+        MembershipSettingsParentFingerprint.result?.let { parentResult ->
+            MembershipSettingsFingerprint.also {
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/music/layout/premium/HideGetPremiumPatch.kt:94]: Is it possible to consolidate this into a single fingerprint? There should be enough to fingerprint.

**Review by oSumAtrIX — COMMENTED** [review]
From what I saw, the patch makes one unnecessary patch at a certain location, but I can't remember which one it was. Toggling the separate patches (There's three places code is patched) may reveal it.

File: src/main/kotlin/app/revanced/util/BytecodeUtils.kt [hunk 3]
```diff
@@ -65,6 +65,17 @@ fun MutableMethod.injectHideViewCall(
     "invoke-static { v$viewRegister }, $classDescriptor->$targetMethod(Landroid/view/View;)V"
 )
 
+/**
+ * Find the index of the first wide literal instruction with the given value.
+ *
+ * @return the first literal instruction with the value, or -1 if not found.
+ */
+fun Method.getWideLiteralInstructionIndex(literal: Long) = implementation?.let {
```
[the remark this answers is not in the document]
Comment (ponces) [reply src/main/kotlin/app/revanced/util/BytecodeUtils.kt]: Done. Thanks.

File: src/main/kotlin/app/revanced/patches/music/layout/premium/HideGetPremiumResourcePatch.kt [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import app.revanced.patcher.patch.ResourcePatch
+import app.revanced.patcher.patch.annotation.Patch
+import app.revanced.patches.shared.misc.mapping.ResourceMappingPatch
+
+@Patch(dependencies = [ResourceMappingPatch::class])
+object HideGetPremiumResourcePatch : ResourcePatch() {
+    internal var privacyTosFooterId = -1L
+
+    override fun execute(context: ResourceContext) {
+        privacyTosFooterId = ResourceMappingPatch.resourceMappings.single {
```
[the remark this answers is not in the document]
Comment (ponces) [reply src/main/kotlin/app/revanced/patches/music/layout/premium/HideGetPremiumResourcePatch.kt]: Done. Thanks.

File: src/main/kotlin/app/revanced/patches/music/layout/premium/HideGetPremiumPatch.kt (hunk 2 above)
Comment (ponces) [reply src/main/kotlin/app/revanced/patches/music/layout/premium/HideGetPremiumPatch.kt]: Done. Thanks

File: src/main/kotlin/app/revanced/patches/music/layout/premium/HideGetPremiumPatch.kt (hunk 1 above)
Comment (ponces) [reply src/main/kotlin/app/revanced/patches/music/layout/premium/HideGetPremiumPatch.kt]: Apparently this is not even needed... Removed. Thanks

**Review by oSumAtrIX — APPROVED** [review]

**Pull request merged**